### PR TITLE
[risk=no][RW-4598] Remove some obsolete Profile fields

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

This is only a portion of RW-4598.  Significant portions are blocked on other stories like https://precisionmedicineinitiative.atlassian.net/browse/RW-6882

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/profile/ProfileMapper.java [hunk 1]
```diff
@@ -31,7 +31,6 @@
       UserMapper.class
     })
 public interface ProfileMapper {
-  @Mapping(target = "contactEmailFailure", ignore = true) // I don't think we actually use this
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/profile/ProfileMapper.java:34]: correct!

File: api/src/test/java/org/pmiops/workbench/actionaudit/auditors/ProfileAuditorTest.kt [hunk 2]
```diff
@@ -63,7 +63,7 @@ class ProfileAuditorTest {
         argumentCaptor<List<ActionAuditEvent>>().apply {
             verify(mockActionAuditService).send(capture())
             val sentEvents: List<ActionAuditEvent> = firstValue
-            assertThat(sentEvents).hasSize(13)
+            assertThat(sentEvents).hasSize(ProfileTargetProperty.values().size)
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/actionaudit/auditors/ProfileAuditorTest.kt:66]: this was not true before I completed the set by adding address below

File: api/src/main/java/org/pmiops/workbench/db/model/DbUser.java [hunk 3]
```diff
@@ -72,7 +72,6 @@
 
   // potentially obsolete profile-style fields (not used)
 
-  @Deprecated private String phoneNumber;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/model/DbUser.java:75]: other obsolete DbUser fields are still referenced elsewhere.  soon...

File: api/src/bigquerytest/java/org/pmiops/workbench/api/CohortReviewControllerBQTest.java [hunk 4]
```diff
@@ -60,7 +60,6 @@
 import org.pmiops.workbench.model.CohortChartDataListResponse;
 import org.pmiops.workbench.model.CohortReview;
 import org.pmiops.workbench.model.Domain;
-import org.pmiops.workbench.model.EmailVerificationStatus;
```
Comment (jmthibault79) [inline api/src/bigquerytest/java/org/pmiops/workbench/api/CohortReviewControllerBQTest.java:63]: Also removed all references to the obsolete `EmailVerificationStatus`

**Comment by jmthibault79** [comment]
Perf tests consume the Profile API so here is a Perf test update: https://github.com/broadinstitute/mcnulty/pull/262

**Review by petesantos — APPROVED** [review]
Sorry for the delay, thanks for the cleanup. LGTM

**Pull request merged**